index.js

```diff
@@ -1,95 +1,69 @@
-      ? `@babel/plugin-${TransformAsyncToGenerator}`
-      : `babel-plugin-${TransformAsyncToGenerator}`;
+      checker.for("ember-maybe-import-regenerator-for-testing").exists() ||
+      checker.for("ember-maybe-import-regenerator").exists();
   },
 
-  included() {
+  included(parent) {
     this._super.included.apply(this, arguments);
-    const app = this._findApp();
-    const babelAddon = this.project.addons.find(
-      addon => addon.name === 'ember-cli-babel'
+    const babelAddon = this.parent.addons.find(
+      addon => addon.name === "ember-cli-babel"
     );
     if (!babelAddon) {
       this.ui.writeWarnLine(
-        '[ember-async-to-generator] Could not find `ember-cli-babel` addon, opting out of transpilation'
+        "[ember-async-to-generator] Could not find `ember-cli-babel` addon, opting out of transpilation"
       );
       return;
     }
-    if (!babelAddon.isPluginRequired(TransformAsyncToGenerator)) {
+    if (!babelAddon.isPluginRequired("transform-async-to-generator")) {
       this.ui.writeInfoLine(
-        `[ember-async-to-generator] ${TransformAsyncToGenerator} is not needed, opting out of transpilation`
+        `[ember-async-to-generator] ${PluginName} is not needed, opting out of transpilation`
       );
       return;
     }
 
-    // Now add the plugin with config to app's babel config
-    app.options = app.options || {};
-    app.options.babel = app.options.babel || {};
-    const plugins = (app.options.babel.plugins =
-      app.options.babel.plugins || []);
-    const hasAsyncPlugin = plugins.find(p =>
-      p[0].match(new RegExp(TransformAsyncToGenerator))
-    );
-    if (hasAsyncPlugin) {
+    // Now add the plugin with config to parent's babel config
+    if (hasPlugin(parent, PluginName)) {
       this.ui.writeWarnLine(
-        `[ember-async-to-generator] you have already listed ${
-          this._asyncPluginName
-        } in babel plugins`
+        `[ember-async-to-generator] ${parent.name} have already listed ${PluginName} in babel plugins`
       );
     } else {
       const moduleMethodOption = {
-        module: 'ember-async-to-generator/helper',
-        method: 'asyncToGenerator'
+        module: "ember-async-to-generator/helper",
+        method: "asyncToGenerator"
       };
-      if (this._isBabel7) {
-        plugins.push([this._asyncPluginName, moduleMethodOption]);
-      } else {
-        const moduleMethodPluginName = 'transform-async-to-module-method';
-        const hasModuleMethodPlugin = plugins.find(p =>
-          p[0].match(new RegExp(moduleMethodPluginName))
-        );
-        if (!hasModuleMethodPlugin) {
-          plugins.push([moduleMethodPluginName, moduleMethodOption]);
-        } else {
-          this.ui.writeWarnLine(
-            `[ember-async-to-generator] you have already listed ${moduleMethodPluginName} in babel plugins`
-          );
-        }
-      }
+      addPlugin(parent, PluginName, moduleMethodOption);
```

Comment:
This bug is fixed in https://github.com/xg-wang/ember-async-to-generator/pull/182